In [2]:
import pandas as pd
pd.set_option('display.max_columns',None)
pd.set_option('display.max_rows',None)

In [3]:
merg1 = pd.read_csv(r"C:\Users\meet.gajera\OneDrive\Desktop\hcdr__\Data\merged\merged1.csv")
prev = pd.read_csv(r"C:\Users\meet.gajera\OneDrive\Desktop\hcdr__\Data\featured\prev_app_featured.csv")

In [4]:
merg1.head()

,Unnamed: 0,SK_ID_CURR,TARGET,CODE_GENDER,CNT_CHILDREN,CNT_FAM_MEMBERS,NAME_CONTRACT_TYPE,AMT_INCOME_TOTAL,AMT_CREDIT,AMT_ANNUITY,AMT_GOODS_PRICE,NAME_INCOME_TYPE,OCCUPATION_TYPE,OCCUPATION_TYPE_MISSING,NAME_EDUCATION_TYPE,NAME_FAMILY_STATUS,FLAG_OWN_CAR,FLAG_OWN_REALTY,NAME_HOUSING_TYPE,FLAG_EMAIL,FLAG_WORK_PHONE,AGE,YEARS_EMPLOYED,YEARS_CURRENT_PHONE,ORG_GROUP,CREDIT_INCOME_RATIO,ANNUITY_INCOME_RATIO,CREDIT_ANNUITY_RATIO,GOODS_CREDIT_RATIO,INCOME_PER_FAM_MEMBER,EMPLOYMENT_AGE_RATIO,IS_SINGLE_PARENT,HAS_DEPENDENTS,HAS_BUREAU_HISTORY,AMT_CREDIT_SUM_sum,AMT_CREDIT_SUM_mean,AMT_CREDIT_SUM_max,AMT_CREDIT_SUM_DEBT_sum,AMT_CREDIT_SUM_DEBT_mean,AMT_CREDIT_SUM_DEBT_MISSING_sum,AMT_CREDIT_SUM_DEBT_MISSING_mean,AMT_CREDIT_SUM_LIMIT_sum,AMT_CREDIT_SUM_LIMIT_MISSING_sum,AMT_CREDIT_SUM_LIMIT_MISSING_mean,DEBT_RATIO_mean,DEBT_RATIO_max,CREDIT_UTILIZATION_mean,HAS_DEBT_sum,IS_ACTIVE_sum,IS_CLOSED_sum,IS_SOLD_sum,IS_BAD_DEBT_sum,YEARS_CREDIT_min,YEARS_CREDIT_max,YEARS_CREDIT_mean
0,0,100002,1,M,0,1.0,Cash loans,202500.0,406597.5,24700.5,351000.0,Working,Laborers,0,Secondary / secondary special,Single / not married,N,Y,House / apartment,0,0,25,1.75,3.11,Business & Corporate,2.0079,0.1220,16.4611,0.8633,202500.0,0.0700,0,0,1,865055.565,108131.945625,450000.0,245781.0,30722.625,3.0,0.375,31988.565,4.0,0.5,0.068275,0.5462,104133.500,1.0,2.0,6.0,0.0,0.0,0.282192,3.936986,2.394521
1,1,100003,0,F,0,2.0,Cash loans,270000.0,1293502.5,35698.5,1129500.0,State servant,Core staff,0,Higher education,Married,N,N,House / apartment,0,0,45,3.25,2.27,Education,4.7908,0.1322,36.2341,0.8732,135000.0,0.0722,0,0,1,1017400.500,254350.125000,810000.0,0.0,0.000,0.0,0.000,810000.000,0.0,0.0,0.000000,0.0000,51850.375,0.0,1.0,3.0,0.0,0.0,1.660274,7.084932,3.837671
2,2,100004,0,M,0,1.0,Revolving loans,67500.0,135000.0,6750.0,135000.0,Working,Laborers,0,Secondary / secondary special,Single / not married,Y,Y,House / apartment,0,1,52,0.62,2.23,Government,2.0000,0.1000,20.0000,1.0000,67500.0,0.0119,0,0,1,189037.800,94518.900000,94537.8,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,94518.900,0.0,0.0,2.0,0.0,0.0,1.117808,3.632877,2.375342
3,3,100006,0,F,0,2.0,Cash loans,135000.0,312682.5,29686.5,297000.0,Working,Laborers,0,Secondary / secondary special,Civil marriage,N,Y,House / apartment,0,0,52,8.33,1.69,Business & Corporate,2.3162,0.2199,10.5328,0.9498,67500.0,0.1602,0,0,0,0.000,0.000000,0.0,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,0.000,0.0,0.0,0.0,0.0,0.0,0.821918,5.005479,2.876712
4,4,100007,0,M,0,1.0,Cash loans,121500.0,513000.0,21865.5,513000.0,Working,Core staff,0,Secondary / secondary special,Single / not married,N,Y,House / apartment,0,0,54,8.32,3.03,Other,4.2222,0.1800,23.4616,1.0000,121500.0,0.1541,0,0,1,146250.000,146250.000000,146250.0,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,146250.000,0.0,0.0,1.0,0.0,0.0,3.147945,3.147945,3.147945


In [5]:
prev.head()

,SK_ID_CURR,PREV_APP_COUNT,APPROVAL_RATE
0,100001,1,1.0
1,100002,1,1.0
2,100003,3,1.0
3,100004,1,1.0
4,100005,2,0.5


In [6]:
merg1['SK_ID_CURR'].isin(prev['SK_ID_CURR']).mean() * 100

94.64929709831517

confirms that about 94.65% of applicants have previous application information and about 5.35% do not.

In [7]:
merge_df2 = merg1.merge(
    prev,
    on='SK_ID_CURR',
    how='left'
)

## Post-Merge Missing Value Handling

In [8]:
merge_df2.isnull().sum().sort_values(ascending=False).head(2)

APPROVAL_RATE     16454
PREV_APP_COUNT    16454
dtype: int64

In [9]:
merge_df2['PREV_APP_COUNT'] = (
    merge_df2['PREV_APP_COUNT']
    .fillna(0)
    .astype(int)
)

merge_df2['APPROVAL_RATE'] = merge_df2['APPROVAL_RATE'].fillna(0)

In [10]:
print(merge_df2.shape)
print(merge_df2.isnull().sum().sort_values(ascending=False).head(20))
print(merge_df2.duplicated().sum())

(307511, 57)
Unnamed: 0                           0
INCOME_PER_FAM_MEMBER                0
IS_SINGLE_PARENT                     0
HAS_DEPENDENTS                       0
HAS_BUREAU_HISTORY                   0
AMT_CREDIT_SUM_sum                   0
AMT_CREDIT_SUM_mean                  0
AMT_CREDIT_SUM_max                   0
AMT_CREDIT_SUM_DEBT_sum              0
AMT_CREDIT_SUM_DEBT_mean             0
AMT_CREDIT_SUM_DEBT_MISSING_sum      0
AMT_CREDIT_SUM_DEBT_MISSING_mean     0
AMT_CREDIT_SUM_LIMIT_sum             0
AMT_CREDIT_SUM_LIMIT_MISSING_sum     0
AMT_CREDIT_SUM_LIMIT_MISSING_mean    0
DEBT_RATIO_mean                      0
DEBT_RATIO_max                       0
CREDIT_UTILIZATION_mean              0
HAS_DEBT_sum                         0
IS_ACTIVE_sum                        0
dtype: int64
0


In [11]:
merge_df2.columns

Index(['Unnamed: 0', 'SK_ID_CURR', 'TARGET', 'CODE_GENDER', 'CNT_CHILDREN',
       'CNT_FAM_MEMBERS', 'NAME_CONTRACT_TYPE', 'AMT_INCOME_TOTAL',
       'AMT_CREDIT', 'AMT_ANNUITY', 'AMT_GOODS_PRICE', 'NAME_INCOME_TYPE',
       'OCCUPATION_TYPE', 'OCCUPATION_TYPE_MISSING', 'NAME_EDUCATION_TYPE',
       'NAME_FAMILY_STATUS', 'FLAG_OWN_CAR', 'FLAG_OWN_REALTY',
       'NAME_HOUSING_TYPE', 'FLAG_EMAIL', 'FLAG_WORK_PHONE', 'AGE',
       'YEARS_EMPLOYED', 'YEARS_CURRENT_PHONE', 'ORG_GROUP',
       'CREDIT_INCOME_RATIO', 'ANNUITY_INCOME_RATIO', 'CREDIT_ANNUITY_RATIO',
       'GOODS_CREDIT_RATIO', 'INCOME_PER_FAM_MEMBER', 'EMPLOYMENT_AGE_RATIO',
       'IS_SINGLE_PARENT', 'HAS_DEPENDENTS', 'HAS_BUREAU_HISTORY',
       'AMT_CREDIT_SUM_sum', 'AMT_CREDIT_SUM_mean', 'AMT_CREDIT_SUM_max',
       'AMT_CREDIT_SUM_DEBT_sum', 'AMT_CREDIT_SUM_DEBT_mean',
       'AMT_CREDIT_SUM_DEBT_MISSING_sum', 'AMT_CREDIT_SUM_DEBT_MISSING_mean',
       'AMT_CREDIT_SUM_LIMIT_sum', 'AMT_CREDIT_SUM_LIMIT_MISSING_sum',
    

In [12]:
merge_df2.head()

,Unnamed: 0,SK_ID_CURR,TARGET,CODE_GENDER,CNT_CHILDREN,CNT_FAM_MEMBERS,NAME_CONTRACT_TYPE,AMT_INCOME_TOTAL,AMT_CREDIT,AMT_ANNUITY,AMT_GOODS_PRICE,NAME_INCOME_TYPE,OCCUPATION_TYPE,OCCUPATION_TYPE_MISSING,NAME_EDUCATION_TYPE,NAME_FAMILY_STATUS,FLAG_OWN_CAR,FLAG_OWN_REALTY,NAME_HOUSING_TYPE,FLAG_EMAIL,FLAG_WORK_PHONE,AGE,YEARS_EMPLOYED,YEARS_CURRENT_PHONE,ORG_GROUP,CREDIT_INCOME_RATIO,ANNUITY_INCOME_RATIO,CREDIT_ANNUITY_RATIO,GOODS_CREDIT_RATIO,INCOME_PER_FAM_MEMBER,EMPLOYMENT_AGE_RATIO,IS_SINGLE_PARENT,HAS_DEPENDENTS,HAS_BUREAU_HISTORY,AMT_CREDIT_SUM_sum,AMT_CREDIT_SUM_mean,AMT_CREDIT_SUM_max,AMT_CREDIT_SUM_DEBT_sum,AMT_CREDIT_SUM_DEBT_mean,AMT_CREDIT_SUM_DEBT_MISSING_sum,AMT_CREDIT_SUM_DEBT_MISSING_mean,AMT_CREDIT_SUM_LIMIT_sum,AMT_CREDIT_SUM_LIMIT_MISSING_sum,AMT_CREDIT_SUM_LIMIT_MISSING_mean,DEBT_RATIO_mean,DEBT_RATIO_max,CREDIT_UTILIZATION_mean,HAS_DEBT_sum,IS_ACTIVE_sum,IS_CLOSED_sum,IS_SOLD_sum,IS_BAD_DEBT_sum,YEARS_CREDIT_min,YEARS_CREDIT_max,YEARS_CREDIT_mean,PREV_APP_COUNT,APPROVAL_RATE
0,0,100002,1,M,0,1.0,Cash loans,202500.0,406597.5,24700.5,351000.0,Working,Laborers,0,Secondary / secondary special,Single / not married,N,Y,House / apartment,0,0,25,1.75,3.11,Business & Corporate,2.0079,0.1220,16.4611,0.8633,202500.0,0.0700,0,0,1,865055.565,108131.945625,450000.0,245781.0,30722.625,3.0,0.375,31988.565,4.0,0.5,0.068275,0.5462,104133.500,1.0,2.0,6.0,0.0,0.0,0.282192,3.936986,2.394521,1,1.000000
1,1,100003,0,F,0,2.0,Cash loans,270000.0,1293502.5,35698.5,1129500.0,State servant,Core staff,0,Higher education,Married,N,N,House / apartment,0,0,45,3.25,2.27,Education,4.7908,0.1322,36.2341,0.8732,135000.0,0.0722,0,0,1,1017400.500,254350.125000,810000.0,0.0,0.000,0.0,0.000,810000.000,0.0,0.0,0.000000,0.0000,51850.375,0.0,1.0,3.0,0.0,0.0,1.660274,7.084932,3.837671,3,1.000000
2,2,100004,0,M,0,1.0,Revolving loans,67500.0,135000.0,6750.0,135000.0,Working,Laborers,0,Secondary / secondary special,Single / not married,Y,Y,House / apartment,0,1,52,0.62,2.23,Government,2.0000,0.1000,20.0000,1.0000,67500.0,0.0119,0,0,1,189037.800,94518.900000,94537.8,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,94518.900,0.0,0.0,2.0,0.0,0.0,1.117808,3.632877,2.375342,1,1.000000
3,3,100006,0,F,0,2.0,Cash loans,135000.0,312682.5,29686.5,297000.0,Working,Laborers,0,Secondary / secondary special,Civil marriage,N,Y,House / apartment,0,0,52,8.33,1.69,Business & Corporate,2.3162,0.2199,10.5328,0.9498,67500.0,0.1602,0,0,0,0.000,0.000000,0.0,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,0.000,0.0,0.0,0.0,0.0,0.0,0.821918,5.005479,2.876712,9,0.555556
4,4,100007,0,M,0,1.0,Cash loans,121500.0,513000.0,21865.5,513000.0,Working,Core staff,0,Secondary / secondary special,Single / not married,N,Y,House / apartment,0,0,54,8.32,3.03,Other,4.2222,0.1800,23.4616,1.0000,121500.0,0.1541,0,0,1,146250.000,146250.000000,146250.0,0.0,0.000,0.0,0.000,0.000,0.0,0.0,0.000000,0.0000,146250.000,0.0,0.0,1.0,0.0,0.0,3.147945,3.147945,3.147945,6,1.000000


In [13]:
merge_df2.drop(columns='Unnamed: 0', inplace=True)

In [14]:
merge_df2.to_csv(r"C:\Users\meet.gajera\OneDrive\Desktop\hcdr__\Data\merged\final_merge.csv", index=False)